sdk/src/main/java/com/qonversion/android/sdk/internal/di/misc/MiscAssemblyImpl.kt

```diff
@@ -0,0 +1,216 @@
+        get() = provideDelayCalculator()
+
+    override val errorResponseMapper: ErrorResponseMapper
+        get() = provideErrorResponseMapper()
+
+    override val appLifecycleObserver: AppLifecycleObserver
+        get() = provideAppLifecycleObserver()
+
+    override val userMapper: UserMapper
+        get() = provideUserMapper()
+
+    override val userPurchaseMapper: UserPurchaseMapper
+        get() = provideUserPurchaseMapper()
+
+    override val entitlementMapper: EntitlementMapper
+        get() = provideEntitlementMapper()
+
+    override val productMapper: ProductMapper
+        get() = provideProductMapper()
+
+    override val subscriptionMapper: SubscriptionMapper
+        get() = provideSubscriptionMapper()
+
+    override val userPropertiesMapper: UserPropertiesMapper
+        get() = provideUserPropertiesMapper()
+
+    override fun init(application: Application) {
+        this.application = application
+    }
+
+    override fun getApiInteractor(retryPolicy: RetryPolicy): ApiInteractor {
+        return provideApiInteractor(retryPolicy)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideInternalConfig(): InternalConfig {
+        return InternalConfig
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideLogger(): Logger {
+        return ConsoleLogger(internalConfig)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideRequestConfigurator(): RequestConfigurator {
+        return RequestConfiguratorImpl(
+            headerBuilder,
+            BASE_API_URL,
+            internalConfig,
+            internalConfig
+        )
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideHeaderBuilder(): HeaderBuilder {
+        return HeaderBuilderImpl(
+            localStorage,
+            locale,
+            internalConfig,
+            internalConfig,
+            internalConfig
+        )
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideSharedPreferences(): SharedPreferences {
+        return application.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideLocalStorage(): LocalStorage {
+        return SharedPreferencesStorage(sharedPreferences)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideLocale(): Locale = Locale.getDefault()
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideNetworkClient(): NetworkClient {
+        return NetworkClientImpl(serializer)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideSerializer(): Serializer = JsonSerializer()
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideDelayCalculator(): RetryDelayCalculator {
+        return ExponentialDelayCalculator(Random)
+    }
```

Comment:
this one too. We should name property and a provide method with an interface name only if there could be a single Implementation. But if there could be several implementations then each of them should have it's own property